.github/workflows/app-artifacts-mac.yml

```diff
@@ -35,22 +41,106 @@ jobs:
+    - name: Download artifact
+      uses: actions/download-artifact@v4
+      with:
+        name: dmgs
+        path: ./dmgs
+    - name: Fetch certificates
+      if: ${{ inputs.signBinaries }}
+      shell: pwsh
+      run: |
+        az login --service-principal -u ${{ secrets.WINDOWS_CLIENT_ID }} -p ${{ secrets.AZ_LOGIN_PASS }}  --tenant 72f988bf-86f1-41af-91ab-2d7cd011db47
+        az keyvault secret download --subscription ${{ secrets.AZ_SUBSCRIPTION_ID }} --vault-name headlamp --name HeadlampAuthCert --file c:\HeadlampAuthCert.pfx --encoding base64
+        az keyvault secret download --subscription ${{ secrets.AZ_SUBSCRIPTION_ID }} --vault-name headlamp --name ESRPHeadlampReqCert --file c:\HeadlampReqCert.pfx --encoding base64
+    - name: Set up certificates
+      if: ${{ inputs.signBinaries }}
+      shell: pwsh
+      run: |
+        Import-PfxCertificate -FilePath c:\HeadlampAuthCert.pfx -CertStoreLocation Cert:\LocalMachine\My -Exportable
+        Import-PfxCertificate -FilePath c:\HeadlampReqCert.pfx -CertStoreLocation Cert:\LocalMachine\My -Exportable
+    - name: Download and Set up ESRPClient
+      if: ${{ inputs.signBinaries }}
+      shell: pwsh
+      run: |
+        nuget.exe sources add -name esrp -source ${{ secrets.ESRP_NUGET_INDEX_URL }} -username headlamp -password ${{ secrets.AZ_DEVOPS_TOKEN }}
+        nuget.exe install Microsoft.EsrpClient -Version 1.2.87 -source  ${{ secrets.ESRP_NUGET_INDEX_URL }} | out-null
+    - name: Sign App
+      shell: pwsh
+      run: |
+        $env:ESRP_PATH="$(Get-Location)\Microsoft.EsrpClient.1.2.87\tools\EsrpClient.exe"
+        $env:HEADLAMP_WINDOWS_CLIENT_ID="${{ secrets.WINDOWS_CLIENT_ID }}"
+        $env:HEADLAMP_WINDOWS_SIGN_EMAIL="${{ secrets.WINDOWS_SIGN_EMAIL }}"
+        cd ./app/scripts
+        node ./esrp.js apple-sign ../../dmgs/
+    - name: Notarize App
+      shell: pwsh
+      run: |
+        $env:ESRP_PATH="$(Get-Location)\Microsoft.EsrpClient.1.2.87\tools\EsrpClient.exe"
+        $env:HEADLAMP_WINDOWS_CLIENT_ID="${{ secrets.WINDOWS_CLIENT_ID }}"
+        $env:HEADLAMP_WINDOWS_SIGN_EMAIL="${{ secrets.WINDOWS_SIGN_EMAIL }}"
+        cd ./app/scripts
+        node ./esrp.js apple-notarize ../../dmgs/
+    - name: Upload Notarized
+      uses: actions/upload-artifact@v4
+      with:
+        name: dmgs
+        path: ./dmgs/Headlamp*.dmg
+        if-no-files-found: error
+        overwrite: true
+        retention-days: 2
+  stapler:
+    runs-on: macos-latest
+    needs: notarize
+    if: ${{ inputs.signBinaries }}
+    steps:
+    - name: Download artifact
+      uses: actions/download-artifact@v4
+      with:
+        name: dmgs
+        path: ./dmgs
+    - name: Staple
+      run: |
+        xcrun stapler staple ./dmgs/Headlamp*.dmg
+    - name: Upload Stapled
+      uses: actions/upload-artifact@v4
+      with:
+        name: dmgs
+        path: ./dmgs/Headlamp*.dmg
+        if-no-files-found: error
+        overwrite: true
+        retention-days: 2
```

Comment:
Should this be only 2 days? If it's the last step, maybe it should stick around longer.

Reply:
Before we had 1 day 🙂 
This should be fine since this action is mostly used for quick testing or for the generation of the release artifacts, in which case I do upload them to the long term release storage.